torchtext/datasets/qqp.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+from torchtext._internal.module_utils import is_module_available
+from torchtext.data.datasets_utils import _create_dataset_directory, _wrap_split_argument
+from typing import Union, Tuple
+
+if is_module_available("torchdata"):
+    from torchdata.datapipes.iter import FileOpener, IterableWrapper
+    from torchtext._download_hooks import HttpReader
+
+URL = "http://qim.fs.quoracdn.net/quora_duplicate_questions.tsv"
+
+MD5 = "b6d5672bd9dc1e66ab2bb020ebeafb8d"
+
+_PATH = "quora_duplicate_questions.tsv"
+
+NUM_LINES = {"train": 404290}
+
+DATASET_NAME = "QQP"
+
+
+@_create_dataset_directory(dataset_name=DATASET_NAME)
+def QQP(root: str):
+    """QQP dataset
+    For additional details refer to https://quoradata.quora.com/First-Quora-Dataset-Release-Question-Pairs
+
+    Args:
+        root: Directory where the datasets are saved. Default: os.path.expanduser('~/.torchtext/cache')
+
+    :returns: DataPipe that yields rows from CoLA dataset (idx (int), label (int), question1 (str), question2 (str))
+    :rtype: str
+    """
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError(
+            "Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`"
+        )
+
+    url_dp = IterableWrapper([URL])
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH),
+        hash_dict={os.path.join(root, _PATH): MD5},
+        hash_type="md5",
+    )
+    cache_dp = HttpReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_dp = FileOpener(cache_dp, encoding="utf-8")
+    # some context stored at top of the file needs to be removed
+    parsed_data = cache_dp.parse_csv(skip_lines=1, delimiter="\t").map(lambda x: (int(x[0]), int(x[-1]), x[3], x[4]))
```

Comment:
I haven't fetched your PR to test. It seems you might need to add `encoding="utf-8" to `parse_csv`.

Reply:
Thanks for having a look! Actually, the issue is visible with STS-B, not QQP, my bad. See https://github.com/pytorch/text/pull/1704.
The extra encoding argument did not solve the issue.